app/controllers/posts_controller.rb

```diff
@@ -7,5 +7,19 @@ def index
   def show
     @post = Post.find(params[:id])
     @comments = @post.comments
+    @like = Like.new
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @user = current_user
+    @post = Post.new(author_id: @user, title: params[:post][:title], text: params[:post][:text])
+    @post.author_id = @user.id
+    @post.save
+    redirect_to user_posts_path(@user)
```

Comment:
- [Optional] Dear @lawrahkonwea, though your implementation is good, a better way to do this would be to check if the save fails and redirect based on the `success` and `failer` conditions. 
> Something like...
```ruby
    def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```
- [Optional] Also, it would be nice if you could make use of the `params method` to filter your params before using them. [Find more here.](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters)